packages/remix-react/browser.tsx

```diff
@@ -189,6 +189,17 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
     });
   }, [location]);
 
+  // Check if __remixRouteModules is missing the module for the current route
+  // and if so, load it.
+  for (let match of router.state.matches) {
+    if (!window.__remixRouteModules[match.route.id]) {
+      window.location.reload();
+      // `reload` executes asynchronously, meaning the error will still flash.
+      // To get around that we return an empty div here.
+      return <div></div>;
```

Comment:
@brophdawg11 this prevents a flash of the error I'm trying to work around. It still produces a set of console errors around problems hydrating the page (no surprise), which seem undesirable. If you have any thoughts on a more holistic way of dealing with this I'd appreciate your feedback. At the same time, this is a rare edge case, so maybe the brief error state resolved by the page reload is acceptable.

Reply:
```suggestion
      return <div/>;
```